### PR TITLE
introducing chatGPT support

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hi @ctrueden and, @acardona , 

this PR adds OpenAI's chatGPT integration into the Script Editor.  Just write a prompt, select it and click on this menu:

image

E.g. producing ImageJ Macro:

image

It also works with Groovy if this language is selected:

image

If this gets merged, we need to:
* [ ] Update documentation to tell people how to use it and how to [configure an OpenAI API Key in the system environment](https://www.immersivelimit.com/tutorials/adding-your-openai-api-key-to-system-environment-variables)
* [ ] Include these libraries and potential dependencies to upload to the ImageJ update site: https://github.com/scijava/script-editor/blob/51b7d2aabee5b05c93f44cad10462a251fe02bc6/pom.xml#L212-L221

Let me know what you think!

Best,
Robert

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -144,6 +144,7 @@
 		<package-name>org.scijava.ui.swing.script</package-name>
 		<license.licenseName>bsd_2</license.licenseName>
 		<license.copyrightOwners>SciJava developers.</license.copyrightOwners>
+		<scijava.app.directory>C:/programs/fiji-win64/Fiji.app/</scijava.app.directory>
```
Comment (haesleinhuepf) [inline pom.xml:147]:
```suggestion
```

File: src/main/java/org/scijava/ui/swing/script/TextEditor.java [hunk 2]
```diff
@@ -106,32 +106,8 @@
...
-import javax.swing.JSplitPane;
-import javax.swing.JTabbedPane;
-import javax.swing.JTextArea;
-import javax.swing.JTextField;
-import javax.swing.JTextPane;
-import javax.swing.JTree;
-import javax.swing.KeyStroke;
-import javax.swing.SwingUtilities;
-import javax.swing.UIManager;
+import javax.swing.*;
```
[the remark this answers is not in the document]
Comment (haesleinhuepf) [reply src/main/java/org/scijava/ui/swing/script/TextEditor.java:109]: Sorry, I think IntelliJ did  this automatically

**Comment by ctrueden** [comment]
I added the Edit>Options>OpenAI... settings dialog. Gonna merge this now, so that we can get it into the next Fiji release. Better exception handling can be added later if/when users have issues with flaky responses from ChatGPT.

**Comment by ctrueden** [comment]
Thanks again, @haesleinhuepf :smile_cat:

File: src/main/java/org/scijava/ui/swing/script/TextEditor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (stefanhahmann) [reply src/main/java/org/scijava/ui/swing/script/TextEditor.java:109]: There is a setting in IntelliJ to change this behaviour: 
grafik

[the remark this answers is not in the document]
Comment (alanocallaghan) [reply src/main/java/org/scijava/ui/swing/script/TextEditor.java:109]: I *think* the relevant setting(s) to change in this screenshot are "[Class/Names] count to use [static] import with '*'", or at least, you should also change those as well as the highlighted one.

**Pull request merged**